tasks/celery.yml

```diff
@@ -0,0 +1,24 @@
+---
+- name: Drop celery config file
+  copy:
+    content: |
+      {% for key in django_stack_celery_svc_conf.keys() %}
+      CELERYD_{{key|upper}}="{{django_stack_celery_svc_conf[key]}}"
+      {% endfor %}
+      CELERYD_CHDIR="{{ active_deploy_dir }}"
+      CELERY_BIN="{{ django_stack_venv_dir }}/bin/celery"
+      CELERY_APP="{{ django_stack_app_name }}"
+    dest: /etc/default/celeryd
+
+- name: Stop existing celery runner
+  service:
+    name: "{{ django_stack_celery_svc_name }}"
+    state: stopped
+  # The celery runner might not exist yet, lets not fail in that scenario
+  ignore_errors: true
```

Comment:
Why do we need to stop it if it's running? Modifying the service template on-disk won't affect the running service until the service is restarted—which it will be via handler notification.

Reply:
Instead I think the role should enforce state=running and enabled=yes on the celery service at the end of configuration. Imagine: no changes are made to the celery config, but the service was stopped manually on the server—as currently written, the role will *not* start celery under those conditions, but it absolutely should.

Reply:
'doh !!! you are right this needs a conditional. 

> Modifying the service template on-disk won't affect the running service until the service is restarted

the larger problem is this though - i want to stop the existing service before i make underlying changes to the systemd unit file and reload it. This was that beta/alpha issue I was hitting. Say I have :  
* a celery worker service already running that references the alpha virtualenv and code-base
* new code is deployed making `beta` the new primary app code. the ansible logic reaches this stage, makes a systemd template change, and now the celery worker has started reference the beta base. This means that the alpha celery worker is still going in the background - which is bad.

> the role will not start celery under those conditions, but it absolutely should.

Shit you are totally right . Nice catch!! this part needs a refactor :(


The